orchestration/OrchestrationService/CustomScriptExecution.ps1

```diff
@@ -0,0 +1,257 @@
+Class CustomScriptExecution {
+
+    hidden $scriptType = "";
+    hidden $command = "";
+    hidden $arguments = @{};
+    
+    # Used to store results from the script execution
+    $Result = "";
+    
+    CustomScriptExecution([string] $command, [hashtable] $arguments) {
+        # Derive the script type from the command being
+        # passed
+        $this.scriptType = `
+            $this.GetScriptType($command);
+        $this.command = $command;
+        $this.arguments += $arguments;
+    }
+
+    [void] Execute() {
+
+        # Branch the execution based on the type of the script being
+        # passed for execution.
+        switch ($this.scriptType.ToLower()) {
+            "powershellscript" {
+                $this.Result = $this.RunPowerShellScript("script");
```

Comment:
What is the reason behind setting the variable result instead of doing: return?, Executing a custom script must return a value, the return can be empty, or contain values.

Reply:
Removed all class variables. The method will instead return the values on call.